src/libraries/System.Private.CoreLib/src/System/String.Comparison.cs

```diff
@@ -199,21 +69,20 @@ public static int Compare(string? strA, string? strB, bool ignoreCase)
         // for meaning of different comparisonType.
         public static int Compare(string? strA, string? strB, StringComparison comparisonType)
         {
+            ThrowIfStringComparisonInvalid(comparisonType);
```

Comment:
Presumably it was written the way it was because now if strA and strB aren't equal and aren't null, we'll be doing an extra unnecessary comparison. No?  Is the smaller code this produces better than having the additional default switch case?

Reply:
It removes `case StringComparison.OrdinalIgnoreCase:` and replaces it with `default:` as that's the only one it can be as the test was done upfront.

One inlined call (just `if` check -> throw helper) seems better than having 3 inlined calls to `CheckStringComparison` plus an explicit throw in the `switch`?

Reply:
Ok.  If it improves size and doesn't harm throughput, sounds good.